src/main/scala/scorex/core/transaction/box/Box.scala

```diff
@@ -6,14 +6,9 @@ import scorex.core.transaction.box.proposition.Proposition
 /**
   * Box is a state element locked by some proposition.
   */
-trait Box[P <: Proposition] extends BytesSerializable {
-  val value: Box.Amount
+trait Box[P <: Proposition, T] extends BytesSerializable {
```

Comment:
What if, instead of making `Box` more generic by changing the type of `value` from `long` to `T` and carrying `T` around as a type parameter, we made `Box` more generic simply by removing `value` entirely?   It would be up to subclasses of `Box` to have fields that contain content (e.g. values or anything else one might want to add to the box).

@kushti @catena2w @zihehuang , what do you think?

Reply:
I know this is a bit old, but that could likely work just fine based on our discussion for #23 . @zihehuang probably had the same mindset I did when we were working on this locally. It's then easy enough to create a parameterised `ValueBox` if you wanted that field, anyway.